src/vm/amd64/unixasmhelpers.S

```diff
@@ -304,7 +232,7 @@ NESTED_ENTRY OnHijackInteriorPointerTripThread, _TEXT, NoHandler
 
     END_PROLOGUE
 
-    call                C_FUNC(OnHijackInteriorPointerWorker)
+    call                C_FUNC(OnHijackWorker)
 
     free_stack          0x8
     pop_register        rax
```

Comment:
You are right - `RDX` and `RAX` are pushed here.

There may be a different problem: Is it possible to have a stack that contains both GC reference and floating point value (xmm0/xmm1)? OnHijackGCTripThread is not preserving floating point return values.

Do we really need both Scalar and GC asm helpes? One universal one should be enough. This is not performance critical path - having one universal helper and smaller/simpler code is better. The only case that needs non-standard hijack helper is x86 floating point return value to deal with floating point stack. Everything else can use the universal one.